test-network-function/lifecycle/suite.go

```diff
@@ -300,6 +300,22 @@ func shutdownTest(podNamespace, podName string) {
 	test.RunAndValidate()
 }
 
+func testNodeDrain(env *config.TestEnvironment, nodeName string) {
+	// drain node
+	drainNode(nodeName)
+	// Ensure the node is uncordoned before exiting the function,
+	// and all deployments are ready
+	defer func() {
+		uncordonNode(nodeName)
+		for _, ns := range env.NameSpacesUnderTest {
+			waitForAllDeploymentsReady(ns, scalingTimeout, scalingPollingPeriod)
```

Comment:
Need to check if the deployments actually become ready again. If not, I think we can abort the test run. Otherwise it would just create more problematic test execution logs for us to triage with no real benefit

Reply:
How can we do that? waitForAllDeploymentsReady() will fail if some deployment is not ready, so the rest of the defer function will not be executed.

I see there's ginkgo.AbortSuite() in 2.0.0 [1], but we're still in pre-2.0. Even calling panic() would just fail this test, but nothing else.

[1] - https://github.com/onsi/ginkgo/issues/734

Reply:
How about extracting this inline function out and have another defer in that funciton?